modules/dxp/apps/portal-workflow/portal-workflow-kaleo-metrics-integration/src/main/java/com/liferay/portal/workflow/kaleo/metrics/integration/internal/helper/IndexerHelper.java

```diff
@@ -95,6 +106,57 @@
 		return localizationMap;
 	}
 
+	public List<Assignment> toAssignments(
+		List<KaleoTaskAssignmentInstance> kaleoTaskAssignmentInstances) {
+
+		List<Assignment> assignments = new ArrayList<>();
+
+		if (ListUtil.isEmpty(kaleoTaskAssignmentInstances)) {
+			return assignments;
+		}
+
+		Stream.of(
+			kaleoTaskAssignmentInstances
+		).flatMap(
+			List::stream
+		).findFirst(
+		).ifPresent(
+			kaleoTaskAssignmentInstance -> {
+				if (Objects.equals(
+						kaleoTaskAssignmentInstance.getAssigneeClassName(),
+						Role.class.getName())) {
+
+					Stream.of(
+						kaleoTaskAssignmentInstances
+					).flatMap(
+						List::stream
+					).collect(
+						Collectors.groupingBy(
+							KaleoTaskAssignmentInstance::getAssigneeClassPK,
+							Collectors.mapping(
+								KaleoTaskAssignmentInstance::getGroupId,
+								Collectors.toList()))
+					).forEach(
+						(assignmentId, assignmentGroupIds) -> assignments.add(
+							new com.liferay.portal.workflow.metrics.model.Role(
+								assignmentGroupIds, assignmentId))
+					);
+				}
+				else {
+					User user = _userLocalService.fetchUser(
+						kaleoTaskAssignmentInstance.getAssigneeClassPK());
```

Comment:
Seems redundant?

https://github.com/liferay-workflow/liferay-portal/pull/495/commits/07c0d49c969f935ae8952173d207b6a86f5fb163#diff-9b34744678450cbd53fd8a0379360f8980f58ac39df7ee05d35c8119bc19a386R565

what do I do, I'll make sure the username will always been define on List<Assignment> or always must get username in this point?